src/PaymentGateways/Stripe/Models/AccountDetail.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Give\PaymentGateways\Stripe\Models;
+
+use InvalidArgumentException;
+
+/**
+ * Class AccountDetail
+ *
+ * @package Give\PaymentGateways\Stripe\Models
+ * @unreleased
+ */
+class AccountDetail {
+	public $type               = null;
+	public $accountName        = null;
+	public $accountSlug        = null;
+	public $accountEmail       = null;
+	public $accountCountry     = null;
+	public $accountId          = null;
+	public $liveSecretKey      = null;
+	public $livePublishableKey = null;
+	public $testSecretKey      = null;
+	public $testPublishableKey = null;
+
+	/**
+	 * Return AccountDetail model
+	 *
+	 * @unreleased
+	 * @param array $args
+	 *
+	 * @return AccountDetail
+	 */
+	public static function fromArray( $args ) {
+		$accountDetail = new static();
+		$accountDetail->validate( $args );
+
+		$accountDetail->type               = $args['type'];
```

Comment:
I think we can reduce the overhead of this class by extending an [ArrayObject](https://www.php.net/manual/en/class.arrayobject.php), in which case the only definition required is the list of required keys used by `validate`.

Reply:
@kjohnson Can you show me how? I did not understand you.

Reply:
Actually, since we aren't iterating on it, maybe  not ArrayObject.

But, it might make sense to use the magic method `__get()` to map properties to array keys.

```php
class AccountDetail {
  protected $args;
  
  public function __construct( $args ) {
    $this->args = $args;
  }
  
  public function __get( $key ) {
    return $this->args[ $key ];
  }
}
```

Reply:
Expanded to include validation.

```php
class AccountDetail {
  protected $args;
  protected $requiredArgs;
  
  public function __construct( $args ) {
    $this->args = $args;
    $this->validate();
  }
  
  public function __get( $key ) {
    return $this->args[ $key ];
  }

  public function validate() {
    // Compare $args to $requiredArgs
  }
}
```